core/src/main/java/io/aiven/kafka/tieredstorage/iceberg/AvroSchemaRegistryStructureProvider.java

```diff
@@ -66,8 +66,20 @@ public void configure(final Map<String, ?> configs) {
         valueDeserializer.configure(serdeConfig, false);
     }
 
-    public ParsedSchema getSchemaById(final int schemaId) throws RestClientException, IOException {
-        return valueDeserializer.getSchemaRegistryClient().getSchemaById(schemaId);
+    public SchemaAndId<Schema> getSchemaById(final Integer schemaId) throws IOException {
+        if (schemaId == null) {
+            return new SchemaAndId(Schema.createUnion(Schema.create(Schema.Type.BYTES),
+                    Schema.create(Schema.Type.NULL)), null);
+        }
+        try {
+            return new SchemaAndId(
+                    Schema.createUnion(Schema.create(Schema.Type.NULL),
+                            (Schema) valueDeserializer.getSchemaRegistryClient().getSchemaById(schemaId).rawSchema()),
+                    schemaId);
+        } catch (final RestClientException e) {
```

Comment:
Don't generic network errors fall into this category?

Reply:
I believe yes but AFAIU it will be thrown only after exhausting all the allowed retry attempts. Or am I missing anything?